### PR TITLE
Replace deprecated spin_until_future_complete

### PR DESCRIPTION
**Pull request by hliberacki**
Replace deprecated spin_until_future_complete with spin_until_complete

Signed-off-by: Hubert Liberacki <hliberacki@gmail.com>

### CODE REVIEW

**Comment by hliberacki** [comment]
Due to change in RCLCPP - https://github.com/ros2/rclcpp/pull/1874 Pull request

File: tf2_tools/tf2_tools/view_frames.py [hunk 1]
```diff
@@ -48,7 +48,7 @@ def main():
     parser.add_argument(
         '--wait-time', '-t', type=float, default=5.0,
         help='Listen to the /tf topic for this many seconds before rendering the frame tree')
-    parser.add_argument('-o','--output', help='Output filename')
+    parser.add_argument('-o', '--output', help='Output filename')
```
Comment (fujitatomoya) [inline tf2_tools/tf2_tools/view_frames.py:51]: i think that we can keep this.

File: tf2_tools/tf2_tools/view_frames.py [hunk 2]
```diff
@@ -83,20 +84,21 @@ def main():
         node.get_logger().error('Service call failed %r' % (e,))
     else:
         node.get_logger().info(
-            'Result:'+ str(result) )
+            'Result:' + str(result))
```
Comment (fujitatomoya) [inline tf2_tools/tf2_tools/view_frames.py:86]: i think that we can keep this.

File: tf2_tools/tf2_tools/view_frames.py [hunk 3]
```diff
@@ -83,20 +84,21 @@ def main():
...
         data = yaml.safe_load(result.frame_yaml)
-        
+
```
Comment (fujitatomoya) [inline tf2_tools/tf2_tools/view_frames.py:88]: i think that we can keep this.

File: tf2_tools/tf2_tools/view_frames.py [hunk 4]
```diff
@@ -83,20 +84,21 @@ def main():
...
         if parsed_args.output is not None:
             frames_gv = '{:s}.gv'.format(parsed_args.output)
             frames_pdf = '{:s}.pdf'.format(parsed_args.output)
         else:
             datetime = time.strftime('%Y-%m-%d_%H.%M.%S')
             frames_gv = 'frames_{:s}.gv'.format(datetime)
             frames_pdf = 'frames_{:s}.pdf'.format(datetime)
-        
+
```
Comment (fujitatomoya) [inline tf2_tools/tf2_tools/view_frames.py:96]: i think that we can keep this.

File: tf2_tools/tf2_tools/view_frames.py [hunk 5]
```diff
@@ -105,6 +107,7 @@ def main():
         rclpy.shutdown()
         return ret
 
+
```
Comment (fujitatomoya) [inline tf2_tools/tf2_tools/view_frames.py:108]: i think that we can keep this.

**Review by fujitatomoya — CHANGES_REQUESTED** [review]
re-formatting changes should be avoided.

**Comment by hliberacki** [comment]
https://github.com/ros2/rclcpp/pull/1874#issuecomment-1086906375 Passing CI with all related PRs linked and build together.

**Review by SteveMacenski — APPROVED** [review]

**Comment by christophebedard** [comment]
I've opened #667 to replace this PR.

**Comment by christophebedard** [comment]
This can be closed, see https://github.com/ros2/geometry2/pull/667#issuecomment-2035715786

**Pull request closed without merging**